website_snippet_carousel_product/static/src/js/snippet.options.js

```diff
@@ -0,0 +1,114 @@
+                this.$target.attr(
+                    "data-products-per-slide",
+                    $elm.attr("data-products-per-slide")
+                );
+                this._refreshTargetSnippet();
+            } else if ($elm.has("[data-products-limit]")) {
+                this.$target.attr(
+                    "data-products-limit",
+                    $elm.attr("data-products-limit")
+                );
+                this._refreshTargetSnippet();
+            }
+            return this._super.apply(this, arguments);
+        },
+
+        /**
+         * @override
+         */
+        onBuilt: function () {
+            var self = this;
+            this._super();
+            this.select_domain().fail(function () {
+                self.getParent()._onRemoveClick($.Event("click"));
+            });
+        },
+
+        /**
+         * @override
+         */
+        _setActive: function () {
+            var self = this;
+            this._super.apply(this, arguments);
+            // Active 'Limit' option
+            this.$el
+                .find("[data-products-limit]")
+                .addBack("[data-products-limit]")
+                .removeClass("active")
+                .filter(function () {
+                    var limit = $(this).data("productsLimit");
+                    var old_limit = self.$target.data("productsLimit") || 12;
+                    return old_limit === limit;
+                })
+                .addClass("active");
+            // Active 'Limit' option
+            this.$el
+                .find("[data-products-per-slide]")
+                .addBack("[data-products-per-slide]")
+                .removeClass("active")
+                .filter(function () {
+                    var limit = $(this).data("productsPerSlide");
+                    var old_limit = self.$target.data("productsPerSlide") || 4;
+                    return old_limit === limit;
+                })
+                .addClass("active");
+        },
+
+        /**
+         * Open domain selector dialog
+         * @returns {Promise}
+         */
+        select_domain: function () {
+            var self = this;
+            var def = wUtils.prompt({
+                id: this.popup_template_id,
+                window_title: this.popup_title,
+                input: _t("Domain (can be empty)"),
+                init: function () {
+                    return self.$target.attr("data-domain");
+                },
+            });
+            def.then(function (domain) {
+                self.$target.attr("data-domain", domain.replace("'", '"'));
+                self._refreshTargetSnippet();
+            });
+            return def;
+        },
+
+        /**
+         * Reset target snippet
+         */
+        _refreshTargetSnippet: function () {
+            this.trigger_up("animation_start_demand", {
+                editableMode: true,
+                $target: this.$target,
+            });
+        },
+    });
```

Comment:
I'm missing the `cleanForSave` method.

You probably don't want the products to be saved in the template code, since they are supposed to change dynamically.

IMHO  you should clean all the code and leave just the container that will trigger the animation.